src/slice.c

```diff
@@ -352,11 +352,11 @@ static SEXP chr_as_index(SEXP i, SEXP x) {
 
 SEXP vec_as_index(SEXP i, SEXP x) {
   switch (TYPEOF(i)) {
+  case NILSXP: return lgl_as_index(Rf_ScalarLogical(1), x);
```

Comment:
This needs protection. Perhaps in `nil_as_index()` so the protection/unprotection doesn't mess up the switch.

Reply:
(I don't want to rely on the fact thet scalarlogical returns a shared constant)

Oh this can just be `vctrs_shared_true`.